psi4/src/psi4/libscf_solver/rhf.cc

```diff
@@ -937,5 +937,34 @@ bool RHF::stability_analysis() {
     // FOLLOW is not implemented for RHF
     return false;
 }
+
+std::shared_ptr<RHF> RHF::c1_deep_copy(std::shared_ptr<BasisSet> basis)
+{
+    std::shared_ptr<Wavefunction> wfn = Wavefunction::c1_deep_copy(basis);
+    auto hf_wfn = std::make_shared<RHF>(wfn, functional_, wfn->options(), wfn->psio());
+    // now just have to copy the matrices that RHF initializes
+    // include only those that are not temporary (some deleted in finalize())
+    if (Ca_) hf_wfn->Ca_ = Ca_subset("AO", "ALL");
+    hf_wfn->Cb_ = hf_wfn->Ca_;
+    if (Da_) hf_wfn->Da_ = Da_subset("AO");
+    hf_wfn->Db_ = hf_wfn->Da_;
+    hf_wfn->D_  = hf_wfn->Da_;
+    if (Fa_) hf_wfn->Fa_ = Fa_subset("AO");
+    hf_wfn->Fb_ = hf_wfn->Fa_;
+    if (epsilon_a_) hf_wfn->epsilon_a_ = 
+        epsilon_subset_helper(epsilon_a_, nsopi_, "AO", "ALL");
+    hf_wfn->epsilon_b_ = hf_wfn->epsilon_a_;
+    // H_ ans X_ reset in the HF constructor, copy them over here
+    SharedMatrix SO2AO = aotoso()->transpose();
+    if (H_) hf_wfn->H_->remove_symmetry(H_, SO2AO);
+    if (X_) hf_wfn->X_->remove_symmetry(X_, SO2AO);
+
+    // pretty sure the next 2 things are really set already
+    same_a_b_dens_ = true;
+    same_a_b_orbs_ = true;
```

Comment:
Dont these need to be set on the returned Wavefunction? If people are blindly copying could be good to do `hf_wfn-> same_a_b_orbs_ = same_a_b_orbs_`.

Reply:
Good point.  Actually these are already set in the RHF constructor so I'm deleting these lines.